nixos/modules/tasks/network-interfaces-scripted.nix

```diff
@@ -54,7 +54,16 @@ let
     };
 
   normalConfig = {
-
+    systemd.network.links = let
+      createNetworkLink = i: nameValuePair "40-${i.name}" {
+        matchConfig.OriginalName = i.name;
```

Comment:
Is this the right behaviour for when users rename their links using udev rules? I'd expect `networking.interfaces.custom.mtu = 1400` to work if I have a rule renaming some interface to "custom", and I think that would have worked with the old script way?

Reply:
`.link` files can only match by original interface names, mac adresses and other properties, not the name to be renamed to - It's probably not even defined how "[legacy udev rules](https://wiki.debian.org/NetworkInterfaceNames#THE_.22PERSISTENT_NAMES.22_SCHEME)" will work together with `.link` (and in what order they're executed)

Considering even debian encourages people to use `.link` files for doing these things, or to just use the predictable network interface names as they are, I'm inclined to add something like this to the release notes:

> Extra care needs to be taken in the presence of [legacy udev rules](https://wiki.debian.org/NetworkInterfaceNames#THE_.22PERSISTENT_NAMES.22_SCHEME) to rename interfaces, as MAC Address and MTU defined in these options can only match on the original link name. In such cases, you most likely want to write a [`.link`](https://www.freedesktop.org/software/systemd/man/systemd.link.html#) file by yourself.

Note this is only a problem with the non-networkd approach - in the case of networkd, MAC Address and MTU are set in the `[Link]` section of the `.network` file via networkd, after renaming happened. As we don't have networkd in scripted networking, we can't use the renamed network interfaces.

Reply:
I kept mentioning `.link` files, as IMHO it's important to understand the concept, but I updated the last phrase to show a more actionable recommendation/alternative, which does link to the corresponding module option, PTAL.